src/Language/Parser.hs

```diff
@@ -113,17 +113,17 @@ parserExtension =
   where
     parserSf :: Parser (Int -> Extension)
     parserSf =
-      do sf <- oneOf "b#"
-         return $ case sf of
-                    'b' -> flat
-                    '#' -> sharp
+      do sf <- string "b" <||> string "#" <|> string "add"
```

Comment:
I _think_ you don't need `<||>` here; `<|>` should be find, since none of these alternatives share a prefix.

Reply:
You are correct:
https://hackage.haskell.org/package/parsec-3.1.14.0/docs/Text-Parsec.html#v:-60--124--62-

Edwin and I generally fell into this pattern so that we didn't have to think about shared prefixes. I think redundancy like this is all over the parser code. We should change it. But we should recognize that we won't correct all these redundancies before moving to a parser generator.

Reply:
I fixed all the other places `<||>` wasn't necessary weeks ago when I refactored the parser, so there shouldn't be anything else like this. You're right though; no reason to worry about it too much.